### ISSUE TITLE
Unnecessary warning when implicitly casting small-enough literal integers to narrower types

### ISSUE THREAD

**Issue by Dentosal** [FuelLabs/sway#606]
This shouldn't produce a warning, as the compiler should know that the literal is small enough.

### Input

```rust
script;
fn main() {
  let a: u32 = 1;
  let b: u32 = a + 1;
}
```

### Expected output

No warnings about casts

### Actual output

```
warning
 --> fuel_project/src/main.sw:0:16
  |
...
2 |   let a: u32 = 1;
  |                - This cast, from integer type of width sixty four to integer type of width thirty two, will lose precision.
  |
warning
 --> fuel_project/src/main.sw:0:20
  |
...
3 |   let b: u32 = a + 1;
  |                    - This cast, from integer type of width sixty four to integer type of width thirty two, will lose precision.
  |
  Compiled script "Fuel example project" with 2 warnings.
```

**Comment by mohammadfawaz** [comment]
This should be fixed with this: https://github.com/FuelLabs/sway/pull/711

**Comment by adlerjohn** [comment]
This isn't fixed with arrays. See #2024

**Comment by anton-trunov** [comment]
Fixed with #4829 I believe
